spring-statemachine-core/src/main/java/org/springframework/statemachine/config/configuration/StateMachineFactoryConfiguration.java

```diff
@@ -146,18 +145,10 @@ public boolean isSingleton() {
 
 		@Override
 		public void afterPropertiesSet() throws Exception {
-			// do not continue without configurers, it would not work
-			if (configurers == null || configurers.size() == 0) {
-				throw new BeanDefinitionStoreException(
-						"Cannot configure state machine due to missing configurers. Did you remember to use " +
-						"@EnableStateMachineFactory with a StateMachineConfigurerAdapter.");
-			}
-			for (AnnotationConfigurer<StateMachineConfig<S, E>, StateMachineConfigBuilder<S, E>> configurer : configurers) {
-				Class<?> clazz = configurer.getClass();
-				if (ClassUtils.getUserClass(clazz).getName().equals(clazzName)) {
-					builder.apply(configurer);
-				}
-			}
+			AnnotationConfigurer<StateMachineConfig<S, E>, StateMachineConfigBuilder<S, E>> configurer = 
+			        (AnnotationConfigurer<StateMachineConfig<S, E>, StateMachineConfigBuilder<S, E>>) beanFactory.getBean(Class.forName(clazzName));
+		    builder.apply(configurer);
```

Comment:
We definitely want to keep above `BeanDefinitionStoreException` as that is thrown when user messed up and don't provide `@EnableStateMachineFactory`. Also there is a similar counter part and call chain when working with `@EnableStateMachine` and these same changes need to go there as well.

I need to check what is the situation with having multiple configurers as that is in theory support by a framework but I have a feeling it doesn't actually work or at least not use by anyone. I know that multiple configurers are not documented nor tested in any way.

Reply:
Other part is in `StateMachineConfiguration`. Code is mostly duplicated as I for some reason I didn't create any shared base classes.

Reply:
I think the case of the exception doesn't make sense anymore. The code is only called if you have an @EnableStateMachineFactory. Because that's what's actually triggering this.

Reply:
Hmm, now that you mentioned it, I think you're right. This throws `BeansException` so probably don't need to do further checks for NPE, etc.